client/config/config.go

```diff
@@ -179,6 +179,8 @@ type Config struct {
 	ConsulConfigs map[string]*structsc.ConsulConfig
 
 	// VaultConfig is this Agent's default Vault configuration
+	//
+	// Deprecated: use GetVaultConfigs() instead.
 	VaultConfig *structsc.VaultConfig
```

Comment:
I keep forgetting that I'm not supposed to read from this value. Wit this, my editor screams at me if I try to 😅

Reply:
You and me both. After we've shipped the beta and have a moment to breathe, I'd like to see if we could remove it entirely. It was too gnarly of a refactoring to do while we were also trying to ship the feature :grinning: